include/roboptim/core/function.hh

```diff
@@ -357,6 +369,33 @@ namespace roboptim
       return boost::get<2> (interval);
     }
 
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Fonction cast as.
+    ///
+    /// \tparam ExpectedType type we want to cast the function into
+    template <class ExpectedType>
+    ExpectedType* castInto()
+    {
+      if (asType<ExpectedType>())
+        return reinterpret_cast<ExpectedType*>(this);
```

Comment:
Wouldn't it be way safer to use a `static_cast` here? (the conversion is supposed to be possible after all)